plugins/nodes/src/js/data/MesosMasters.js

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { Observable } from "rxjs/Observable";
+import "rxjs/add/observable/timer";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/retry";
+import "rxjs/add/operator/filter";
+
+import MesosStateStore from "#SRC/js/stores/MesosStateStore";
+
+import LeaderGrid from "../components/LeaderGrid";
+
+function isEmptyObject(obj) {
+  return Object.keys(obj).length === 0;
+}
+
+function hostPort(address) {
+  return `${address.hostname}:${address.port}`;
+}
+
+function supportsRegion(master) {
+  return (
+    master.domain &&
+    master.domain.fault_domain &&
+    master.domain.fault_domain.region &&
+    master.domain.fault_domain.region.name
+  );
+}
+
+export function getRegion(master) {
+  return supportsRegion(master)
+    ? master.domain.fault_domain.region.name
```

Comment:
Somewhat of a nit -- AFAIK, you can do:
`return findNestedPropertyInObject(master, "domain.fault_domain.region.name") || "N/A";`

Reply:
Wow, I love that one. Will take a look. Thanks `findNestedPropertyInObject`

Reply:
Tearing down masters should be a little trick, you can ssh into the master machines (tell me if you need help with that) and then you can use iptables to block all OUTBOUND on port 2181 (which is zookeper port), that disconects the master from election, but does not kill the master.